keps/sig-architecture/4330-compatibility-versions/README.md

```diff
@@ -0,0 +1,1301 @@
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We intend to introduce version compatibility and emulation options to
+Kubernetes control plane components to make upgrades safer by increasing the
+granularity of steps available to cluster administrators. We will introduce
+`--emulation-version` flag to emulate the behavior (APIs, features, ...) of a
+prior Kubernetes release version. We will also introduce a
+`--min-compatibility-version` flag to control which prior versions of Kubernetes
+a control plane component is compatible with (in terms of APIs, features, ...)
+and can be rolled back to.
+
+When a emulation version is provided, the [Version skew policy](https://kubernetes.io/releases/version-skew-policy/)
+will apply to the emulation version instead of the binary version, allowing binary
+versions to be skipped when performing skip version upgrades, so long the
+upgrade transitions incrementally through emulation versions according to the
```

Comment:
How does min-compatibility-version play into this? Is there some relationship between skew and that version too, or not? I guess you'll get to it later...but you might want to just say that here?

Reply:
That's a really good question. I missed this in the edit.  I'll clarify.

Reply:
I'll put this in a section about skew since it's non-trivial and doesn't fit into the summary particularly well.